src/ResourceManagement/Azure/Azure.Tests/RedisCachesTests.cs

```diff
@@ -0,0 +1,228 @@
+        
+        [Fact(Skip = "TODO: Convert to recorded tests")]
+        public void CanCRUDRedisCache()
+        {
+            try
+            {
+                var redisManager = CreateRedisManager();
+
+                // Create
+                var resourceGroups = redisManager.ResourceManager.ResourceGroups
+                                        .Define(RG_NAME_SECOND)
+                                        .WithRegion(Region.US_CENTRAL)
+                                        .Create();
+
+                var redisCacheDefinition1 = redisManager.RedisCaches
+                        .Define(RR_NAME)
+                        .WithRegion(Region.ASIA_EAST)
+                        .WithNewResourceGroup(RG_NAME)
+                        .WithBasicSku()
+                        .Create();
+
+                var redisCacheDefinition2 = redisManager.RedisCaches
+                        .Define(RR_NAME_SECOND)
+                        .WithRegion(Region.US_CENTRAL)
+                        .WithExistingResourceGroup(resourceGroups)
+                        .WithPremiumSku()
+                        .WithShardCount(10)
+                        .WithPatchSchedule(DayOfWeek.Sunday, 10, TimeSpan.FromMinutes(302))
+                        .Create();
+
+                var redisCacheDefinition3 = redisManager.RedisCaches
+                        .Define(RR_NAME_THIRD)
+                        .WithRegion(Region.US_CENTRAL)
+                        .WithExistingResourceGroup(resourceGroups)
+                        .WithPremiumSku(2)
+                        .WithRedisConfiguration("maxclients", "2")
+                        .WithNonSslPort()
+                        .Create();
+                
+                var redisCache = redisCacheDefinition1;
+                var redisCachePremium = redisCacheDefinition3;
+                Assert.Equal(RG_NAME, redisCache.ResourceGroupName);
+                Assert.Equal(SkuName.Basic, redisCache.Sku.Name);
+
+                // List by Resource Group
+                var redisCaches = redisManager.RedisCaches.ListByGroup(RG_NAME);
+                bool found = false;
+                foreach (IRedisCache existingRedisCache in redisCaches)
+                {
+                    if (existingRedisCache.Name.Equals(RR_NAME, StringComparison.OrdinalIgnoreCase))
+                    {
+                        found = true;
+                    }
+                }
+                Assert.True(found);
+                Assert.Equal(1, redisCaches.Count);
+
+                // List all Redis resources
+                redisCaches = redisManager.RedisCaches.List();
+                found = false;
+                foreach (IRedisCache existingRedisCache in redisCaches)
+                {
+                    if (existingRedisCache.Name.Equals(RR_NAME, StringComparison.OrdinalIgnoreCase))
+                    {
+                        found = true;
+                    }
+                }
```

Comment:
May be when we manually convert the unit test/samples we could take advantage of LINQ, like:

var found = redisCaches.List().Any(rc => rc.Name.Equals(RR_NAME, StringComparison.OrdinalIgnoreCase));

CC-ing team for inputs on this specific question @hovsepm  @anudeepsharma @martinsawicki @jianghaolu @alvadb